pkg/isoeditor/rhcos_test.go

```diff
@@ -55,52 +56,152 @@ var _ = Context("with test files", func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 
 		It("missing iso file", func() {
-			editor := NewEditor(workDir)
+			editor := NewEditor(workDir, "")
 			err := editor.CreateMinimalISOTemplate("invalid", testRootFSURL, "x86_64", minimalISOPath)
 			Expect(err).To(HaveOccurred())
 		})
+
+		It("missing nmstatectl binary", func() {
+			editor := NewEditor(workDir, "")
+			err := editor.CreateMinimalISOTemplate(isoFile, testRootFSURL, "x86_64", minimalISOPath)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("cpu arch mismatch", func() {
+			nmstatectl, err := os.CreateTemp(os.TempDir(), "nmstatectl")
+			Expect(err).ToNot(HaveOccurred())
+
+			editor := NewEditor(workDir, nmstatectl.Name())
+			err = editor.CreateMinimalISOTemplate(isoFile, testRootFSURL, "some-arch", minimalISOPath)
+			Expect(err).ToNot(HaveOccurred())
+		})
 	})
 
 	Describe("CreateFCOSMinimalISOTemplate", func() {
 		It("iso created successfully", func() {
-			editor := NewEditor(workDir)
+			nmstatectl, err := os.CreateTemp(os.TempDir(), "nmstatectl")
+			Expect(err).ToNot(HaveOccurred())
+
+			editor := NewEditor(workDir, nmstatectl.Name())
 
-			err := editor.CreateMinimalISOTemplate(isoFile, testFCOSRootFSURL, "x86_64", minimalISOPath)
+			err = editor.CreateMinimalISOTemplate(isoFile, testFCOSRootFSURL, "x86_64", minimalISOPath)
 			Expect(err).ToNot(HaveOccurred())
 		})
 
 		It("missing iso file", func() {
-			editor := NewEditor(workDir)
+			editor := NewEditor(workDir, "")
 			err := editor.CreateMinimalISOTemplate("invalid", testFCOSRootFSURL, "x86_64", minimalISOPath)
 			Expect(err).To(HaveOccurred())
 		})
 	})
-	It("fixGrubConfig alters the kernel parameters correctly", func() {
-		err := fixGrubConfig(testRootFSURL, filesDir)
-		Expect(err).ToNot(HaveOccurred())
 
-		newLine := "	linux /images/pxeboot/vmlinuz random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal 'coreos.live.rootfs_url=%s'"
-		grubCfg := fmt.Sprintf(newLine, testRootFSURL)
-		validateFileContainsLine(filepath.Join(filesDir, "EFI/redhat/grub.cfg"), grubCfg)
+	Describe("createNmstateRamDisk", func() {
+		var (
+			nmstatectlPath, extractDir, arch, ramDiskPath string
+			err                                           error
+		)
+
+		BeforeEach(func() {
+			extractDir, err = os.MkdirTemp(os.TempDir(), "isoutil")
+			Expect(err).ToNot(HaveOccurred())
+
+			nmstatectl, err := os.CreateTemp(extractDir, "nmstatectl")
+			Expect(err).ToNot(HaveOccurred())
+			nmstatectlPath = nmstatectl.Name()
 
-		newLine = "	initrd /images/pxeboot/initrd.img /images/ignition.img %s"
-		grubCfg = fmt.Sprintf(newLine, ramDiskImagePath)
-		validateFileContainsLine(filepath.Join(filesDir, "EFI/redhat/grub.cfg"), grubCfg)
+			ramDisk, err := os.CreateTemp(extractDir, "nmstate.img")
+			Expect(err).ToNot(HaveOccurred())
+			ramDiskPath = ramDisk.Name()
 
+			arch = normalizeCPUArchitecture(runtime.GOARCH)
+		})
+
+		AfterEach(func() {
+			os.Remove(nmstatectlPath)
```

Comment:
You should be removing the entire `extractDir` path here. Otherwise I think you'll end up leaking temp dirs.